app/components/common/collectionSnackBar/index.tsx

```diff
@@ -0,0 +1,143 @@
+        });
+      }
+    },
+    [isOpen, collectionId]
+  );
+
+  return (
+    <Snackbar
+      classes={{ root: s.snackbarWrapper }}
+      anchorOrigin={{
+        vertical: 'bottom',
+        horizontal: 'center',
+      }}
+      open={isOpen}
+      onClose={() => dispatch(closeCollectionSnackBar())}
+      autoHideDuration={5000}
+      ClickAwayListenerProps={{ mouseEvent: false, touchEvent: false }}
+      ContentProps={{
+        'aria-describedby': 'message-id',
+      }}
+      message={
+        <span id="message-id" className={s.snackbarContext}>
+          {`Saved to `}
+          {isLongName && <br />}
+          <Link
+            className={s.collectionName}
+            to={`/collections/${collectionId}`}
+            onClick={() => {
+              ActionTicketManager.trackTicket({
+                pageType: getCurrentPageType(),
+                actionType: 'fire',
+                actionArea: 'collectionSnackbar',
+                actionTag: 'clickCollectionTitle',
+                actionLabel: String(collectionId),
+              });
+
+              dispatch(closeCollectionSnackBar());
+              dispatch(closeDialog());
+            }}
+          >{`${collectionName}.`}</Link>
+        </span>
+      }
+      action={[
+        <div className={s.viewCollectionBtn} key={`viewCollection`}>
+          <Button
+            elementType="link"
+            variant="text"
+            color="blue"
+            size="large"
+            to={`/collections/${collectionId}`}
+            onClick={() => {
+              ActionTicketManager.trackTicket({
+                pageType: getCurrentPageType(),
+                actionType: 'fire',
+                actionArea: 'collectionSnackbar',
+                actionTag: 'clickViewCollection',
+                actionLabel: String(collectionId),
+              });
+
+              dispatch(closeCollectionSnackBar());
+              dispatch(closeDialog());
+            }}
+          >
+            View Collection
+          </Button>
+        </div>,
+        <div className={s.closeBtn} key={`close`}>
+          <Button
+            elementType="button"
+            variant="text"
+            color="gray"
+            size="small"
+            onClick={() => {
+              ActionTicketManager.trackTicket({
+                pageType: getCurrentPageType(),
+                actionType: 'fire',
+                actionArea: 'collectionSnackbar',
+                actionTag: 'clickCloseButton',
+                actionLabel: null,
+              });
+
+              dispatch(closeCollectionSnackBar());
+            }}
+          >
+            <Icon icon="X_BUTTON" />
+          </Button>
+        </div>,
+      ]}
+    />
+  );
+};
+
+const mapStateToProps = (state: AppState) => {
+  return {
+    collectionSnackBarState: state.collectionSnackBarState,
+    layout: state.layout,
+  };
+};
+
+export default connect(mapStateToProps)(CollectionSnackBar);
```

Comment:
there is no need to use `connect` API anymore.
you can use `useSelector` from `react-redux` package.
and it looks more natural using hooks API than HoC pattern in Functional Component.